tests/test_recharge_discovery.py

```diff
@@ -0,0 +1,92 @@
+        • verify that all other fields have inclusion of available metadata.
+        """
+        streams_to_test = self.expected_streams()
+
+        conn_id = connections.ensure_connection(self)
+
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Verify stream names follow naming convention
+        # streams should only have lowercase alphas and underscores
+        found_catalog_names = {c['tap_stream_id'] for c in found_catalogs}
+        self.assertTrue(all([re.fullmatch(r"[a-z_]+",  name) for name in found_catalog_names]),
+                        msg="One or more streams don't follow standard naming")
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+                # Verify ensure the catalog is found for a given stream
+                catalog = next(iter([catalog for catalog in found_catalogs
+                                     if catalog["stream_name"] == stream]))
+                self.assertIsNotNone(catalog)
+
+                # collecting expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_keys = self.expected_replication_keys()[stream]
+                expected_automatic_fields = expected_primary_keys | expected_replication_keys
+
+                # collecting actual values...
+                schema_and_metadata = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+                metadata = schema_and_metadata["metadata"]
+                stream_properties = [item for item in metadata if item.get("breadcrumb") == []]
+                actual_primary_keys = set(
+                    stream_properties[0].get(
+                        "metadata", {self.PRIMARY_KEYS: []}).get(self.PRIMARY_KEYS, [])
+                )
+                actual_automatic_fields = set(
+                    item.get("breadcrumb", ["properties", None])[1] for item in metadata
+                    if item.get("metadata").get("inclusion") == "automatic"
+                )
+
+                ##########################################################################
+                ### metadata assertions
+                ##########################################################################
+
+                # verify there is only 1 top level breadcrumb in metadata
+                self.assertTrue(len(stream_properties) == 1,
+                                msg="There is NOT only one top level breadcrumb for {}".format(stream) + \
+                                "\nstream_properties | {}".format(stream_properties))
+
+                # verify primary key(s) match expectations
+                self.assertSetEqual(expected_primary_keys, actual_primary_keys)
+
+                # verify that primary keys and replication keys
+                # are given the inclusion of automatic in metadata.
+                self.assertSetEqual(expected_automatic_fields, actual_automatic_fields)
+
```

Comment:
Could you please add the following assertions

Verify that if there is a replication key we are doing INCREMENTAL otherwise FULL.
Verify the actual replication matches our expected replication method.

Here is an example https://github.com/singer-io/tap-mixpanel/blob/37e445fcbcae8fcaaf7b8e5adead3717df5799de/tests/tap_tester/test_discovery.py#L90-L103

Reply:
Also here is the up-to-date requirements https://github.com/stitchdata/tap-tester/blob/master/reference/expectations/saas_taps.md

Reply:
added

Reply:
I missed one more assertion in my initial review

Verify the valid-replication-keys match the expected replication keys.

Reply:
@kspeer825 I've added in an assertion for this